vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp

```diff
@@ -1,23 +1,206 @@
+            }
+        });
+        
+   });
+    
+    // ---- Cancel Edited Description ----
+    
+    $("#cancelEditDescription").click(function(){
+        $("#submitDescription").hide()
+        $("#editDescription").show()
+        $("#cancelEditDescription").hide()
+        var description = $("#description").data('value')
+        $('description').insertBefore( "#newDescription" );
+        $("#newDescription").remove()
+        $("#description").show()
+        
+    });
+    
+    // ------- Edit/Save/Cancel Description ends --------
+    
+    
+    
+    //------- edit title of Sequence starts --------
+    
+    $("#submitTitle").hide();
+    $("#cancelEditTitle").hide();
+    
+    $("#editTitle").click(function() {
+    	// Using data attribute
+    	var sequenceTitle = $("#title").data('value')  // gets value without using trim function
+        $('<div class="col-4" id = "editSequenceTitle"><input id="newTitle" class="form-control" type="text"></div>').insertAfter( "#title" );
+    	$("#title").val(sequenceTitle)
+    	$('#title').text('Sequence: ')
+        $("#newTitle").val(sequenceTitle)
+        $("#editTitle").hide()
+        $("#submitTitle").show()
+        $("#cancelEditTitle").show()
+        
+    });
+    	
+    	
+    	
+    // ------ Submit title Sequence -------
+    	
+    $("#submitTitle").click(function() {
+        var formData = new FormData();
+        formData.append('title', $("#newTitle").val());
+        $.ajax({
+            url: "<c:url value="/staff/module/${module.id}/sequence/${sequence.id}/edit/title?${_csrf.parameterName}=${_csrf.token}" />",
+            type: 'POST',
+            cache       : false,
+            contentType : false,
+            processData : false,
+            data: formData,
+            enctype: 'multipart/form-data',
+            success: function(data) {
+                // replace text box with new description
+                $("#submitTitle").hide()
+                $("#cancelEditTitle").hide();
+                $("#editTitle").show()
+                var val = $("#newTitle").val();
+                $("#editSequenceTitle").remove()
+                $("#title").text("Sequence: " + val)
+                $("#title").data('value', val)  // sets value
+            },
+            error: function(data) {
+                var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>We are sorry but something went wrong. Please try again later.</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+                $('.error').append(alert);
+            }
+        });
+                
+      });
+    
+    
+   // ------  Edit Title Sequence ------
+   
+   $("#cancelEditTitle").click(function(){
+        $("#submitTitle").hide()
+        $("#editTitle").show()
+        $("#cancelEditTitle").hide()
+        var sequenceTitle = $("#title").val()
```

Comment:
there is not value attribute for h2

Reply:
if you delete newDescription in line 68, why inserting it here?
-- I am not inserting new Description again. I am just added original description tag before "newDescription" using insertBefore and then deleting "newDescription"

Reply:
why adding the original description again? it's already in your html.

Reply:
h2 does not have a value attribute